service_container.rst

```diff
@@ -690,6 +690,40 @@ argument for *any* service defined in this file! You can bind arguments by name
 The ``bind`` config can be also be applied to specific services or when loading many
 services at once (i.e. :ref:`service-psr4-loader`).
 
+Getting Container Parameters as a Service
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. versionadded:: 4.1
+    The ``ContainerBagInterface`` class was introduced in Symfony 4.1.
+
+If some service or controller needs lots of container parameters, there's an
+easier alternative to binding all of them. Type-hint an argument with the
+:class:`Symfony\\Component\\DependencyInjection\\ParameterBag\\ContainerBagInterface`
+class to inject all container parameters in a single object similar to the
+:class:`Symfony\\Component\\DependencyInjection\\ParameterBag\\ParameterBag`
+objects used in other parts of the framework::
+
+    // src/Service/MessageGenerator.php
```

Comment:
An example of the services.yml/xml would be nice also. How do I configure which services are inside the ContainerBag?

Reply:
In the ContainerBag there are no services, just all container parameters. About the YAML/XML config ... since we're promoting autowiring and type-hint injection, I prefer to keep consistent with the rest of the docs.

Reply:
I know, but how do I say that inside this containerBag there are a param called 'parameter_name' (your example)

Reply:
@jordisala1991 I had the same question as you, read the docs again and now my understanding is that you get each and every available parameters in that bag

Reply:
I've reworded everything. Hopefully it's easier to understand now. Thanks!

Reply:
Now I get it! Thank you @javiereguiluz